components/ChipTextField/examples/MDCChipTextField.m

```diff
@@ -0,0 +1,360 @@
+    _chipContainerViewConstraintTrailingConstant = 0.0f;
+    [self setupChipsContainerView];
+
+    [self addTarget:self
+                  action:@selector(chipTextFieldTextDidChange)
+        forControlEvents:UIControlEventEditingChanged];
+
+    [self addTextFieldObservers];
+  }
+  return self;
+}
+
+- (void)addTextFieldObservers {
+  [[NSNotificationCenter defaultCenter]
+      addObserver:self
+         selector:@selector(textFieldDidEndEditingWithNotification:)
+             name:UITextFieldTextDidEndEditingNotification
+           object:self];
+  [[NSNotificationCenter defaultCenter]
+      addObserver:self
+         selector:@selector(textFieldDidBeginEditingWithNotification:)
+             name:UITextFieldTextDidBeginEditingNotification
+           object:self];
+}
+
+- (void)setupChipsContainerView {
+  MDCChipTextFieldScrollView *chipsContainerView =
+      [[MDCChipTextFieldScrollView alloc] initWithFrame:CGRectZero];
+  chipsContainerView.translatesAutoresizingMaskIntoConstraints = NO;
+  chipsContainerView.chipSpacing = kChipsSpacing;
+  chipsContainerView.dataSource = self;
+  self.chipsContainerView = chipsContainerView;
+  [self addSubview:chipsContainerView];
+
+  NSLayoutConstraint *chipContainerViewConstraintTop =
+      [NSLayoutConstraint constraintWithItem:self.chipsContainerView
+                                   attribute:NSLayoutAttributeTop
+                                   relatedBy:NSLayoutRelationEqual
+                                      toItem:self
+                                   attribute:NSLayoutAttributeTop
+                                  multiplier:1
+                                    constant:0];
+  NSLayoutConstraint *chipContainerViewConstraintBottom =
+      [NSLayoutConstraint constraintWithItem:self.chipsContainerView
+                                   attribute:NSLayoutAttributeBottom
+                                   relatedBy:NSLayoutRelationEqual
+                                      toItem:self
+                                   attribute:NSLayoutAttributeBottom
+                                  multiplier:1
+                                    constant:0];
+  [NSLayoutConstraint
+      activateConstraints:@[ chipContainerViewConstraintTop, chipContainerViewConstraintBottom ]];
+  [self updateChipViewLeadingConstraints];
+  [self updateChipViewTrailingConstraints];
+}
+
+- (void)appendChipWithText:(NSString *)text {
+  MDCChipView *chipView = [[MDCChipView alloc] init];
+  chipView.titleLabel.text = text;
+  chipView.translatesAutoresizingMaskIntoConstraints = NO;
+
+  [self.chipsContainerView appendChipView:chipView];
+
+  // recalculate the layout to get a correct chip frame values
+  [self.chipsContainerView layoutIfNeeded];
+  [self.chipViews addObject:chipView];
+
+  [self clearChipsContainerOffsetWithConstant:kTextToEnterPlaceholderLength];
+}
+
+// TODO: the constant here reflects the margin, places calling this method need to be refactored.
```

Comment:
Is this in our "todo"?

Reply:
It is a code refactor but the comment here is not clear enough. Going to add more explanation here:)

Reply:
Addressed in 0f433959bc04c76f0573eef55b8d2c880d5caea7.